BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -708,6 +709,366 @@ open class SBAStudyParticipantCustomAttributesProfileItem: SBAProfileItemBase {
     }
 }
 
+open class SBAWhatAndWhen: NSObject, Comparable {
+    static var valueKey: String { return #keyPath(value) }
+    static var dateKey: String { return #keyPath(date) }
+    static var isNewKey: String { return #keyPath(isNew) }
+    open var value: SBBJSONValue
+    open var date: NSDate
+    var isNew: Bool
+    
+    public init(dictionaryRepresentation dictionary: [String: SBBJSONValue]) {
+        value = dictionary[SBAWhatAndWhen.valueKey]!
+        let dateString = dictionary[SBAWhatAndWhen.dateKey] as! String
+        date = NSDate(iso8601String: dateString)
+        let isNewJson = dictionary[SBAWhatAndWhen.isNewKey] as? NSNumber
+        isNew = isNewJson != nil ? isNewJson!.boolValue : false
+        super.init()
+    }
+    
+    public init(_ value: SBBJSONValue, asOf date: NSDate, isNew:Bool = false) {
+        self.value = value
+        self.date = date
+        self.isNew = isNew
+        super.init()
+    }
+    
+    public func dictionaryRepresentation() -> [String: SBBJSONValue] {
+        return [
+            SBAWhatAndWhen.valueKey: value,
+            SBAWhatAndWhen.dateKey: date.iso8601String() as NSString
+        ]
+    }
+    
+    public func cachedDictionaryRepresentation() -> [String: SBBJSONValue] {
+        var dict = self.dictionaryRepresentation()
+        dict[SBAWhatAndWhen.isNewKey] = isNew as NSNumber
+        return dict
+    }
+}
+
+public func < (lhs: SBAWhatAndWhen, rhs: SBAWhatAndWhen) -> Bool {
+    return (lhs.date as Date) < (rhs.date as Date)
+}
+
+public func == (lhs: SBAWhatAndWhen, rhs: SBAWhatAndWhen) -> Bool {
+    return (lhs.date as Date) == (rhs.date as Date)
```

Comment:
These are used when adding to a Set to prevent duplication, and when sorting an array by date. I can add value comparison as a secondary check. It's not useful for what I needed these for, but doesn't hurt either.